source/server/admin/utils.cc

```diff
@@ -44,9 +44,8 @@ bool filterParam(Http::Utility::QueryParams params, Buffer::Instance& response,
   auto p = params.find("filter");
   if (p != params.end()) {
     const std::string& pattern = p->second;
-    try {
-      regex = std::regex(pattern);
-    } catch (std::regex_error& error) {
+    TRY_NEEDS_AUDIT { regex = std::regex(pattern); }
```

Comment:
main thread only

Reply:
Yes, this is main thread only.

Reply:
so .. switch to TRY_ASSERT_MAIN_THREAD ?

Reply:
Sure, I have fixed it in my local branch, will push the update later.